web/src/worker.ts

```diff
@@ -0,0 +1,99 @@
+import * as Comlink from "comlink";
+import init, { MapModel } from "backend";
+import type { TravelMode } from "./stores";
+import type { Position, Feature, Polygon, FeatureCollection } from "geojson";
+
+export class Backend {
+  inner: MapModel | null;
+
+  constructor() {
+    this.inner = null;
+  }
+
+  async loadFile(osmBytes: Uint8Array, progressCb: (msg: string) => void) {
+    // TODO Do we need to do this only once?
+    await init();
+
+    this.inner = new MapModel(osmBytes, progressCb);
+  }
+
+  isLoaded(): boolean {
+    return this.inner != null;
+  }
+
+  unset() {
+    this.inner = null;
+  }
+
+  getBounds(): [number, number, number, number] {
+    if (!this.inner) {
+      throw new Error("Backend used without a file loaded");
+    }
+
+    return Array.from(this.inner.getBounds()) as [
```

Comment:
It's a bit verbose to repeat all the Rust APIs again, but it is a nice chance to fix up any TS problems and make the API nicer in the rest of the frontend code